src/platform/tests/MpiServerTests.cpp

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <CommonTests.h>
+#include <CommonUtils.h>
+#include <MpiServer.h>
+#include <Mpi.h>
+
+namespace Tests
+{
+    class MpiServerTests : public ::testing::Test {};
+
+    static const char* g_errorClientName = "Error_Client";
+    static const char* g_errorComponent = "Error_Component";
+    static const char* g_errorObject = "Error_Object";
+    static const char* g_mockHandle = "Mock_Client_Handle";
+    static const char* g_mockPayload = "\"MockPayload\"";
+
+    static MPI_HANDLE MockCallMpiOpen(const char* clientName, const unsigned int maxPayloadSizeBytes)
+    {
+        UNUSED(maxPayloadSizeBytes);
+
+        if (strcmp(clientName, g_errorClientName) == 0)
+        {
+            return nullptr;
+        }
+        else
+        {
+            char* handle = new (std::nothrow) char[strlen(g_mockHandle) + 1];
+            if (handle != nullptr)
+            {
+                strcpy(handle, g_mockHandle);
+            }
+            return (MPI_HANDLE)handle;
+        }
+    }
+
+    static void MockCallMpiClose(MPI_HANDLE handle)
+    {
+        UNUSED(handle);
+    }
+
+    static int MockCallMpiSet(MPI_HANDLE handle, const char* componentName, const char* objectName, MPI_JSON_STRING payload, const int payloadSize)
+    {
+        UNUSED(handle);
+        UNUSED(payload);
+        UNUSED(payloadSize);
+
+        return (0 == strcmp(componentName, g_errorComponent)) && (0 == strcmp(objectName, g_errorObject)) ? -1 : MPI_OK;
+    }
+
+    static int MockCallMpiGet(MPI_HANDLE handle, const char* componentName, const char* objectName, MPI_JSON_STRING* payload, int* payloadSize)
+    {
+        UNUSED(handle);
+
+        if (0 == strcmp(componentName, g_errorComponent) && 0 == strcmp(objectName, g_errorObject))
+        {
+            return -1;
+        }
+        else
+        {
+            *payload = new (std::nothrow) char[strlen(g_mockPayload) + 1];
+            if (*payload != nullptr)
+            {
+                strcpy(*payload, g_mockPayload);
+                *payloadSize = strlen(g_mockPayload);
+            }
+            return MPI_OK;
+        }
+    }
+
+    static int MockCallMpiSetDesired(MPI_HANDLE handle, const MPI_JSON_STRING payload, const int payloadSize)
+    {
+        UNUSED(handle);
+        UNUSED(payloadSize);
+
+        return (0 == strcmp(payload, g_mockPayload)) ? MPI_OK : -1;
+    }
+
+    static int MockCallMpiGetReported(MPI_HANDLE handle, MPI_JSON_STRING* payload, int* payloadSize)
+    {
+        UNUSED(handle);
+
+        *payload = new (std::nothrow) char[strlen(g_mockPayload) + 1];
```

Comment:
>*payload = new (std::nothrow) [](http://example.com/codeflow?start=8&length=29)

Where is this freed?

Reply:
Added `FREE_MEMORY` to each of the tests where memory is allocated.